store/listings/management/commands/fetch_listings.py

```diff
@@ -1,44 +1,79 @@
+import logging
+
+from django.conf import settings
 from django.core.management.base import BaseCommand
+from rest_framework import serializers
+
+from store.common.fields import PriceField
+from store.listings.models import Listing
+from store.listings.utils import ListingsRequest
 
-from listings_api_keys import API_KEY, API_TOKEN
-from store.listings.utils import ListingsRequest, save_to_file
+logger = logging.getLogger('listings')
 
 
 class Command(BaseCommand):
-    help = 'Fetch listings from an external API and create text files'
+    help = 'Fetch listings from an external API and create db objects'
 
     def handle(self, *args, **options):
-        api_key = API_KEY
-        api_token = API_TOKEN
-        hostname = 'https://api.pythonic.me/v1'
-
-        listings_request = ListingsRequest(api_key, api_token, hostname)
+        listings_request = ListingsRequest(
+            api_key=settings.PYTHONIC_API_KEY,
+            api_token=settings.PYTHONIC_API_TOKEN,
+            hostname=settings.PYTHONIC_API_HOSTNAME)
 
         listings_data = [
             {
                 'data': listings_request.get_digital_listings(),
-                'filename': 'digital_listings.txt'
             },
             {
                 'data': listings_request.get_soft_tag_listings(),
-                'filename': 'soft_tags.txt'
             },
             {
                 'data': listings_request.get_usd_listings_under_20(),
-                'filename': 'usd_listings_under_20.txt'
             },
             {
                 'data': listings_request.get_blanket_search_listings(),
-                'filename': 'blanket.txt'
             },
             {
                 'data': listings_request.get_cotton_eur_listings(),
-                'filename': 'cotton_eur.txt'
             },
         ]
 
-        for listings in listings_data:
-            save_to_file(listings=listings)
+        from django.db.transaction import atomic
+        with atomic():
+            for listings in listings_data:
+                for listing in listings['data']:
+                    uid = listing.get('listing_id')
+                    if not uid:
+                        logger.error(f"Missing uid for listing: {listing.get('title')}")
+                        continue
+
+                    price = listing.get('price_amount')
+                    if price is None:
+                        logger.error(f"Missing price for listing: {listing.get('title')}")
+                        continue
+
+                    try:
+                        price = PriceField.validate_price(str(price))
+                    except serializers.ValidationError as e:
+                        logger.error(f"Invalid price for listing: {listing.get('title')}: {e}")
```

Comment:
Using logger is a correct approach, but I would still recommend you to define logger in `/store/common/utils.py` and use it it within your own log() function - gives you more flexibility to change things in the future.

Then, in fetch_listings, instead of using `logger.error`, use your own `log()` function.

Examples how how I would do it - in screenshots



<img width="1031" alt="Screenshot 2024-07-23 at 11 21 51" src="https://github.com/user-attachments/assets/e3d20b2a-a2a0-4004-8bee-1594c93b6f29">

<img width="1004" alt="Screenshot 2024-07-23 at 11 19 49" src="https://github.com/user-attachments/assets/299f771f-d9ce-46bb-8092-0ef727552bf4">

Reply:
The logger is not outputting anything in the console because it requires more configuration. 

You can use the following prompt to figure out the proper config:

> logger = logging.getLogger('listings')
> how to configure logger in python to make it output logs in console as I'm executing the script

But I would still recommend you to use a simple `print()` within your own `log()` function - this is a simple and easy way to have logs in the beginning of the project. 

Logger is typically used for more advanced cases and requires more attention to config, files, etc. For you this seems like an overkill atm.